### ISSUE TITLE
Do not get stuck on Block sync

### ISSUE THREAD

**Issue by NicolasDorier** [ACINQ/eclair#2181]
Version 0.7.0
Related to https://github.com/ACINQ/eclair/issues/2180 as well.

We can't use regtest because even if we restart eclair after it crashes, it fails to sync to the top.
The block height returned by the node is always higher than the block height of eclair's getinfo api, and is stuck there.

**Comment by NicolasDorier** [comment]
The block height returned by the node is always higher than the block height of eclair's getinfo api

**Comment by NicolasDorier** [comment]
No problem in the previous version it seems

**Comment by t-bast** [comment]
Can you detail that? Maybe provide some logs? I'm not sure what the issue is here exactly.

**Comment by NicolasDorier** [comment]
@t-bast 


```
2022-02-16 08:48:39,124 INFO  fr.acinq.eclair.Plugin - loading 0 plugins
2022-02-16 08:48:39,595 INFO  a.e.slf4j.Slf4jLogger - Slf4jLogger started
2022-02-16 08:48:40,551 INFO  fr.acinq.eclair.Setup - hello!
2022-02-16 08:48:40,553 INFO  fr.acinq.eclair.Setup - version=0.7.0 commit=notag
2022-02-16 08:48:40,554 INFO  fr.acinq.eclair.Setup - datadir=/data
2022-02-16 08:48:40,555 INFO  fr.acinq.eclair.Setup - initializing secure random generator
2022-02-16 08:48:40,587 INFO  f.a.eclair.NodeParams - use seed file: /data/node_seed.dat
2022-02-16 08:48:40,588 INFO  f.a.eclair.NodeParams - use seed file: /data/channel_seed.dat
2022-02-16 08:48:40,635 INFO  f.a.secp256k1.Secp256k1 - secp256k1 library successfully loaded
2022-02-16 08:48:40,677 INFO  fr.acinq.eclair.Setup - instanceid=4416a9aa-ab58-4e99-983c-35d20847eb13
2022-02-16 08:48:41,176 INFO  fr.acinq.eclair.Setup - nodeid=0347d29270ae8e47b23755972d3c1efaa488512fbe8926c768493f1241c13139d4 alias=eclair
2022-02-16 08:48:41,176 INFO  fr.acinq.eclair.Setup - using chain=regtest chainHash=06226e46111a0b59caaf126043eb5bbf28c34f3a5e332a1fc7b2b73cf188910f
2022-02-16 08:48:41,914 INFO  fr.acinq.eclair.Setup - current blockchain height=2
2022-02-16 08:48:41,931 INFO  fr.acinq.eclair.Setup - current feeratesPerKB=FeeratesPerKB(5000 sat/kB,210000 sat/kB,180000 sat/kB,150000 sat/kB,110000 sat/kB,50000 sat/kB,20000 sat/kB,15000 sat/kB,5000 sat/kB) feeratesPerKw=FeeratesPerKw(1250 sat/kw,52500 sat/kw,45000 sat/kw,37500 sat/kw,27500 sat/kw,12500 sat/kw,5000 sat/kw,3750 sat/kw,1250 sat/kw)
2022-02-16 08:48:41,993 INFO  f.a.eclair.router.Router - loading network announcements from db...
2022-02-16 08:48:41,998 INFO  f.a.eclair.router.Router - loaded from db: channels=0 nodes=0
2022-02-16 08:48:42,052 INFO  f.a.eclair.router.Router - initialization completed, ready to process messages
2022-02-16 08:48:42,089 INFO  fr.acinq.eclair.Setup - initial wallet address=bcrt1qv3cke5qvetcf4jajzu5v3lulxqmasgv0erd0yn
2022-02-16 08:48:42,183 INFO  akka.actor.ActorCell - channels listener is ready
2022-02-16 08:48:42,211 INFO  f.a.e.p.r.PostRestartHtlcCleaner - htlcsIn=0 notRelayed=0 relayedOut=0
2022-02-16 08:48:42,211 INFO  f.a.e.p.r.PostRestartHtlcCleaner - notRelayed=List()
2022-02-16 08:48:42,211 INFO  f.a.e.p.r.PostRestartHtlcCleaner - relayedOut=Map()
2022-02-16 08:48:42,221 INFO  f.a.e.io.Switchboard - restoring 0 peer(s) with 0 channel(s)
2022-02-16 08:48:42,258 INFO  f.acinq.eclair.io.Server CON - bound on /0.0.0.0:9735
2022-02-16 08:48:43,074 INFO  f.a.e.b.b.zmq.ZMQActor - connected to tcp://172.27.0.2:28333
2022-02-16 08:48:43,074 INFO  f.a.e.b.b.zmq.ZMQActor - connected to tcp://172.27.0.2:28332
2022-02-16 08:48:43,077 INFO  fr.acinq.eclair.Boot - json API enabled on port=8080
2022-02-16 08:48:45,157 INFO  f.a.eclair.router.Router PAY h:95a7350871a537da70ef7f9f6137cc854288c7948dff3358cb0d462d3d066a21 p:7b1247ad i:7b1247ad - finding routes 0347d29270ae8e47b23755972d3c1efaa488512fbe8926c768493f1241c13139d4->02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 with assistedChannels= ignoreNodes= ignoreChannels= excludedChannels=
2022-02-16 08:48:45,158 INFO  f.a.eclair.router.Router PAY h:95a7350871a537da70ef7f9f6137cc854288c7948dff3358cb0d462d3d066a21 p:7b1247ad i:7b1247ad - finding routes with params=RouteParams(false,SearchBoundaries(21000 msat,0.03,6,CltvExpiryDelta(1008)),Left(WeightRatios(0.0,0.05,0.4,0.55,RelayFees(500 msat,200))),MultiPartParams(15000000 msat,5),control,false), multiPart=true
2022-02-16 08:48:45,176 WARN  f.a.e.p.s.MultiPartPaymentLifecycle PAY n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 h:95a7350871a537da70ef7f9f6137cc854288c7948dff3358cb0d462d3d066a21 p:7b1247ad i:7b1247ad - router error: route not found
2022-02-16 08:48:45,267 WARN  f.a.e.p.s.MultiPartPaymentLifecycle PAY n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 h:95a7350871a537da70ef7f9f6137cc854288c7948dff3358cb0d462d3d066a21 p:7b1247ad i:7b1247ad - multi-part payment failed
2022-02-16 08:48:45,276 INFO  a.a.DeadLetterActorRef - Message [fr.acinq.eclair.payment.PaymentFailed] from Actor[akka://eclair-node/user/$k/payment-initiator#-1201093162] to Actor[akka://eclair-node/deadLetters] was not delivered. [1] dead letters encountered. If this is not an expected behavior then Actor[akka://eclair-node/deadLetters] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
2022-02-16 08:48:45,284 INFO  f.a.e.p.s.MultiPartPaymentLifecycle PAY n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 h:95a7350871a537da70ef7f9f6137cc854288c7948dff3358cb0d462d3d066a21 p:7b1247ad i:7b1247ad - failed payment attempts details: {"paymentHash":"95a7350871a537da70ef7f9f6137cc854288c7948dff3358cb0d462d3d066a21","destination":"02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74","totalAmount":1000,"pathFindingExperiment":"control","failures":[{"amount":1000,"route":[],"message":"route not found"}]}
2022-02-16 08:48:45,776 ERROR f.a.eclair.Boot - API call failed with cause=peer 02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 not found
2022-02-16 08:48:45,885 INFO  f.a.e.i.ReconnectionTask CON n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - connecting to eclair_dest/172.27.0.5:9736
2022-02-16 08:48:45,885 INFO  f.a.e.io.ClientSpawner - initiating new connection to nodeId=02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 origin=Actor[akka://eclair-node/deadLetters]
2022-02-16 08:48:45,889 INFO  f.acinq.eclair.io.Client CON n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - connecting to eclair_dest:9736
2022-02-16 08:48:45,896 INFO  f.acinq.eclair.io.Client CON n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - connected to eclair_dest:9736
2022-02-16 08:48:46,009 INFO  f.a.e.io.PeerConnection CON n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - connection authenticated with 02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74@eclair_dest:9736 direction=outgoing
2022-02-16 08:48:46,013 INFO  f.a.e.io.PeerConnection CON n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - using features=option_onion_messages:optional,gossip_queries_ex:optional,option_data_loss_protect:optional,var_onion_optin:mandatory,option_static_remotekey:optional,option_support_large_channel:optional,option_anchors_zero_fee_htlc_tx:optional,payment_secret:mandatory,option_shutdown_anysegwit:optional,option_channel_type:optional,basic_mpp:optional,gossip_queries:optional
2022-02-16 08:48:46,083 INFO  f.a.e.io.PeerConnection CON n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - peer is using features=option_onion_messages:optional,gossip_queries_ex:optional,option_data_loss_protect:optional,var_onion_optin:mandatory,option_static_remotekey:optional,option_support_large_channel:optional,option_anchors_zero_fee_htlc_tx:optional,payment_secret:mandatory,option_shutdown_anysegwit:optional,option_channel_type:optional,basic_mpp:optional,gossip_queries:optional, networks=06226e46111a0b59caaf126043eb5bbf28c34f3a5e332a1fc7b2b73cf188910f
2022-02-16 08:48:46,087 INFO  f.a.e.io.PeerConnection CON n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - not syncing with this peer
2022-02-16 08:48:46,087 INFO  f.a.e.io.PeerConnection CON n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - rebroadcast will be delayed by 25 seconds
2022-02-16 08:48:46,096 INFO  f.a.e.i.ReconnectionTask CON n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - peer is connected
2022-02-16 08:48:46,160 INFO  f.a.eclair.router.Router PAY h:95a7350871a537da70ef7f9f6137cc854288c7948dff3358cb0d462d3d066a21 p:b52b9130 i:b52b9130 - finding routes 0347d29270ae8e47b23755972d3c1efaa488512fbe8926c768493f1241c13139d4->02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 with assistedChannels= ignoreNodes= ignoreChannels= excludedChannels=
2022-02-16 08:48:46,161 INFO  f.a.eclair.router.Router PAY h:95a7350871a537da70ef7f9f6137cc854288c7948dff3358cb0d462d3d066a21 p:b52b9130 i:b52b9130 - finding routes with params=RouteParams(false,SearchBoundaries(21000 msat,0.03,6,CltvExpiryDelta(1008)),Left(WeightRatios(0.0,0.05,0.4,0.55,RelayFees(500 msat,200))),MultiPartParams(15000000 msat,5),control,false), multiPart=true
2022-02-16 08:48:46,161 WARN  f.a.e.p.s.MultiPartPaymentLifecycle PAY n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 h:95a7350871a537da70ef7f9f6137cc854288c7948dff3358cb0d462d3d066a21 p:b52b9130 i:b52b9130 - router error: route not found
2022-02-16 08:48:46,175 WARN  f.a.e.p.s.MultiPartPaymentLifecycle PAY n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 h:95a7350871a537da70ef7f9f6137cc854288c7948dff3358cb0d462d3d066a21 p:b52b9130 i:b52b9130 - multi-part payment failed
2022-02-16 08:48:46,175 INFO  a.a.DeadLetterActorRef - Message [fr.acinq.eclair.payment.PaymentFailed] from Actor[akka://eclair-node/user/$k/payment-initiator#-1201093162] to Actor[akka://eclair-node/deadLetters] was not delivered. [2] dead letters encountered. If this is not an expected behavior then Actor[akka://eclair-node/deadLetters] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
2022-02-16 08:48:46,177 INFO  f.a.e.p.s.MultiPartPaymentLifecycle PAY n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 h:95a7350871a537da70ef7f9f6137cc854288c7948dff3358cb0d462d3d066a21 p:b52b9130 i:b52b9130 - failed payment attempts details: {"paymentHash":"95a7350871a537da70ef7f9f6137cc854288c7948dff3358cb0d462d3d066a21","destination":"02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74","totalAmount":1000,"pathFindingExperiment":"control","failures":[{"amount":1000,"route":[],"message":"route not found"}]}
2022-02-16 08:48:46,419 INFO  fr.acinq.eclair.io.Peer n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - requesting a new channel with type=anchor_outputs_zero_fee_htlc_tx fundingSatoshis=16777215 sat, pushMsat=0 msat and fundingFeeratePerByte=Some(253 sat/kw) temporaryChannelId=c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c localParams=LocalParams(0347d29270ae8e47b23755972d3c1efaa488512fbe8926c768493f1241c13139d4,m/625242397'/1705058930'/1621377549'/474013496/2002631596'/1079108194'/1403009930/1808419653'/1',546 sat,5000000000,167772 sat,1 msat,CltvExpiryDelta(720),30,true,ByteVector(22 bytes, 0x0014e51537ab476f37cbd8536bb4ff8b4bb8274b6bd1),None,option_onion_messages:optional,gossip_queries_ex:optional,option_data_loss_protect:optional,var_onion_optin:mandatory,option_static_remotekey:optional,option_support_large_channel:optional,option_anchors_zero_fee_htlc_tx:optional,payment_secret:mandatory,option_shutdown_anysegwit:optional,option_channel_type:optional,basic_mpp:optional,gossip_queries:optional)
2022-02-16 08:48:46,494 INFO  f.a.e.i.Peer n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 c:c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c - OUT msg=OpenChannel(06226e46111a0b59caaf126043eb5bbf28c34f3a5e332a1fc7b2b73cf188910f,c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c,16777215 sat,0 msat,546 sat,5000000000,167772 sat,1 msat,2500 sat/kw,CltvExpiryDelta(720),30,03b81cfca44ea4ec7ffb210d340703a135f25cfbce066cb0d1545acd1b3d071192,035fd172a5d9bd1b5ee872feb53ce20187f4e29ed86f86f314035b95d701290030,027ec109b5c9fd40cfcd1848d245cbe33e555c2b73c13babb650f83bfa029abbd6,03d1ac7bc387f0493127f7e27f1032862c5c3d2318fa6968accfcaea869267855e,0372621d95a1a64072151ca532bc048923832fdb07757dfdb564836b059837e929,037a0a83b7614586259e0233eb304970b4de3add67829148607cd65016d63e8e0c,ChannelFlags(announceChannel=true),TlvStream(ArraySeq(UpfrontShutdownScriptTlv(ByteVector(empty)), ChannelTypeTlv(anchor_outputs_zero_fee_htlc_tx)),List()))
2022-02-16 08:48:46,495 INFO  f.a.eclair.Diagnostics n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 c:c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c - OUT msg=OpenChannel(06226e46111a0b59caaf126043eb5bbf28c34f3a5e332a1fc7b2b73cf188910f,c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c,16777215 sat,0 msat,546 sat,5000000000,167772 sat,1 msat,2500 sat/kw,CltvExpiryDelta(720),30,03b81cfca44ea4ec7ffb210d340703a135f25cfbce066cb0d1545acd1b3d071192,035fd172a5d9bd1b5ee872feb53ce20187f4e29ed86f86f314035b95d701290030,027ec109b5c9fd40cfcd1848d245cbe33e555c2b73c13babb650f83bfa029abbd6,03d1ac7bc387f0493127f7e27f1032862c5c3d2318fa6968accfcaea869267855e,0372621d95a1a64072151ca532bc048923832fdb07757dfdb564836b059837e929,037a0a83b7614586259e0233eb304970b4de3add67829148607cd65016d63e8e0c,ChannelFlags(announceChannel=true),TlvStream(ArraySeq(UpfrontShutdownScriptTlv(ByteVector(empty)), ChannelTypeTlv(anchor_outputs_zero_fee_htlc_tx)),List()))
2022-02-16 08:48:46,654 INFO  f.a.eclair.Diagnostics n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 c:c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c - IN msg=AcceptChannel(c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c,546 sat,5000000000,167772 sat,1 msat,1,CltvExpiryDelta(720),30,02bb2769cd9436ba79dde52a00abb98a17e029befb26c00b7e2b9f19c1e1621822,0289cd9264759266de3997dfd43580d2886e03b09eb23728a33add1da871f810a6,02548235627526be76fd9db6e08c1e49df4e408113294146014542fe6577b1dd21,0273129cddf1b341ed4f8bd4fd58be225591e1963e1e60a59e55873498c2e62ed9,022664b61eea6af8f220720599ad325023a38804ea98f430723e6db7f0a06913de,03f4cd53c613cf9ad870aa35866121878d75c1be7d8e6a2ea2cb99723524c4e73f,TlvStream(List(UpfrontShutdownScriptTlv(ByteVector(empty)), ChannelTypeTlv(anchor_outputs_zero_fee_htlc_tx)),List()))
2022-02-16 08:48:46,655 INFO  f.a.e.i.Peer n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 c:c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c - IN msg=AcceptChannel(c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c,546 sat,5000000000,167772 sat,1 msat,1,CltvExpiryDelta(720),30,02bb2769cd9436ba79dde52a00abb98a17e029befb26c00b7e2b9f19c1e1621822,0289cd9264759266de3997dfd43580d2886e03b09eb23728a33add1da871f810a6,02548235627526be76fd9db6e08c1e49df4e408113294146014542fe6577b1dd21,0273129cddf1b341ed4f8bd4fd58be225591e1963e1e60a59e55873498c2e62ed9,022664b61eea6af8f220720599ad325023a38804ea98f430723e6db7f0a06913de,03f4cd53c613cf9ad870aa35866121878d75c1be7d8e6a2ea2cb99723524c4e73f,TlvStream(List(UpfrontShutdownScriptTlv(ByteVector(empty)), ChannelTypeTlv(anchor_outputs_zero_fee_htlc_tx)),List()))
2022-02-16 08:48:46,957 INFO  fr.acinq.eclair.io.Peer n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - channel id switch: previousId=c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c nextId=4101c0c1c3e87ae15534426c40bb6e22053d7eeeb70995376019378a4fff2ab5
2022-02-16 08:48:46,958 INFO  f.a.e.i.Peer n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 c:c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c - OUT msg=FundingCreated(c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c,4101c0c1c3e87ae15534426c40bb6e22053d7eeeb70995376019378a4fff2ab4,1,8668ddd8a2977ae50c422e54eb283799bf4eed3b0782f8f880d011fa454a52651abf8a5239b24cbba6b5d76e4e009819f50b94055a5f6ae4b0081e3418e1dbb7,TlvStream(List(),List()))
2022-02-16 08:48:46,958 INFO  f.a.e.io.PeerConnection n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - sending sync channel range query with flags_opt=Some(QueryFlags(3)) replacePrevious=false
2022-02-16 08:48:46,959 INFO  f.a.eclair.Diagnostics n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 c:c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c - OUT msg=FundingCreated(c25db7cb56193886685486ff233d97c052b4be84af608a6bc68e015f4086409c,4101c0c1c3e87ae15534426c40bb6e22053d7eeeb70995376019378a4fff2ab4,1,8668ddd8a2977ae50c422e54eb283799bf4eed3b0782f8f880d011fa454a52651abf8a5239b24cbba6b5d76e4e009819f50b94055a5f6ae4b0081e3418e1dbb7,TlvStream(List(),List()))
2022-02-16 08:48:46,962 INFO  f.a.eclair.router.Router n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - sending query_channel_range=QueryChannelRange(06226e46111a0b59caaf126043eb5bbf28c34f3a5e332a1fc7b2b73cf188910f,BlockHeight(0),2147483647,TlvStream(List(QueryFlags(3)),List()))
2022-02-16 08:48:46,990 INFO  f.a.eclair.router.Router SYN n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - received empty reply_channel_range, sync is complete
2022-02-16 08:48:47,016 INFO  f.a.eclair.router.Router SYN n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - received query_channel_range with firstBlockNum=BlockHeight(0) numberOfBlocks=2147483647 extendedQueryFlags_opt=TlvStream(List(QueryFlags(3)),List())
2022-02-16 08:48:47,017 INFO  f.a.e.io.PeerConnection SYN n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - setting up gossipTimestampFilter=GossipTimestampFilter(06226e46111a0b59caaf126043eb5bbf28c34f3a5e332a1fc7b2b73cf188910f,1645001327 unixsec,2147483647,TlvStream(List(),List()))
2022-02-16 08:48:47,020 INFO  f.a.eclair.router.Router SYN n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 - replying with 0 items for range=(BlockHeight(0), 2147483647)
2022-02-16 08:48:47,034 INFO  f.a.eclair.Diagnostics n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 c:4101c0c1c3e87ae15534426c40bb6e22053d7eeeb70995376019378a4fff2ab5 - IN msg=FundingSigned(4101c0c1c3e87ae15534426c40bb6e22053d7eeeb70995376019378a4fff2ab5,4a1cb176771055a3affc9f34d39a358935f45c5a1a4cbba7d3e1cfd86c6d8f3e0efbaa4cdf12bc8dea32937e818e29656c6e4e8cedf72021812f3f0c6c6e39d1,TlvStream(List(),List()))
2022-02-16 08:48:47,034 INFO  f.a.e.i.Peer n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 c:4101c0c1c3e87ae15534426c40bb6e22053d7eeeb70995376019378a4fff2ab5 - IN msg=FundingSigned(4101c0c1c3e87ae15534426c40bb6e22053d7eeeb70995376019378a4fff2ab5,4a1cb176771055a3affc9f34d39a358935f45c5a1a4cbba7d3e1cfd86c6d8f3e0efbaa4cdf12bc8dea32937e818e29656c6e4e8cedf72021812f3f0c6c6e39d1,TlvStream(List(),List()))
2022-02-16 08:48:47,052 INFO  f.a.e.channel.Channel n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 c:4101c0c1c3e87ae15534426c40bb6e22053d7eeeb70995376019378a4fff2ab5 - publishing funding tx for channelId=4101c0c1c3e87ae15534426c40bb6e22053d7eeeb70995376019378a4fff2ab5 fundingTxid=b42aff4f8a371960379509b7ee7e3d05226ebb406c423455e17ae8c3c1c00141
2022-02-16 08:48:47,053 INFO  f.a.e.channel.Channel n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 c:4101c0c1c3e87ae15534426c40bb6e22053d7eeeb70995376019378a4fff2ab5 - committing txid=b42aff4f8a371960379509b7ee7e3d05226ebb406c423455e17ae8c3c1c00141
2022-02-16 08:48:47,076 INFO  f.a.e.db.DbEventHandler n:02664a616dc11b79e0b911d7ca847768ac8bcc35393f9d524d74f06c6f8343bf74 c:4101c0c1c3e87ae15534426c40bb6e22053d7eeeb70995376019378a4fff2ab5 - paying mining fee=154 sat for txid=b42aff4f8a371960379509b7ee7e3d05226ebb406c423455e17ae8c3c1c00141 desc=funding
```

Eclair is stuck at `height = 2`, while the main chain has 109 blocks.
Mining new blocks doesn't improve the situation.

**Comment by NicolasDorier** [comment]
What is ` using chain=regtest chainHash=06226e46111a0b59caaf126043eb5bbf28c34f3a5e332a1fc7b2b73cf188910f` ? what is this hash? doesn't seem to be in the chain...

**Comment by NicolasDorier** [comment]
ah ok this is the block hash of the genesis in reverse... might it be the issue?

**Comment by NicolasDorier** [comment]
@t-bast I tried to run my tests on 0.6.1 -> no problem, do not get stuck. It fails from 0.6.2.

**Comment by sstone** [comment]
It may be related to https://github.com/ACINQ/eclair/pull/1910: we now subscribe to block hashes, your bitcoin node's configuration must include something like:
```
zmqpubhashblock=tcp://127.0.0.1:29000
```

**Comment by t-bast** [comment]
Right, this is what @sstone mentions. We use the ZMQ block hash subscription, otherwise we just pull from `bitcoind` every 10 minutes. If you simply update your `bitcoin.conf` this will work again.

**Comment by NicolasDorier** [comment]
I confirm, this was the problem! tests pass now!

**Comment by NicolasDorier** [comment]
It would be nice to have some error or something to spot it easily though!
Eclair could easily start.

**Comment by t-bast** [comment]
We do log an error, but only after 10 minutes, because that's when we detect that we haven't received a block via ZMQ...so it's really not ideal for regtest, but works better for testnet/mainnet.

We could list ZMQ subscriptions at start-up though, to verify that bitcoind is correctly configured, I haven't tried the RPCs bitcoind exposes for that yet.
